Homework11/Homework11/Catalog.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Text.Json;
+
+namespace Homework11
+{
+    public class Catalog
+    {
+        public List<Product> Products { get; set; }
+
+        public Catalog()
+        {
+            Products = new List<Product>();
+        }
+
+        public void AddProducts(Product product)
+        {
+            Products.Add(product);
+        }
+
+        public List<string> SerializeJsonDemo()
+        {
+            List<string> serializedProducts = new List<string>();
+
+            foreach (var product in Products)
+            {
+                string serializedJson = JsonSerializer.Serialize(product);
+                serializedProducts.Add(serializedJson);
+            }
+
+            foreach (var serializedProduct in serializedProducts)
+            {
+                Console.WriteLine(serializedProduct);
+            }
```

Comment:
Ten druhy for cyklus je velmi neefektivni. Muzes ten vypis preci udelat uz v ramci toho prvniho for cyklu. Usetris tim jeden pruchod kolekci

```suggestion
            foreach (var product in Products)
            {
                string serializedJson = JsonSerializer.Serialize(product);
                serializedProducts.Add(serializedJson);
                Console.WriteLine(serializedJson);
            }
```